genesyscloud/architect_flow/resource_genesyscloud_architect_flow_proxy.go

```diff
@@ -83,46 +98,143 @@ func (a *architectFlowProxy) GetAllFlows(ctx context.Context, name string, varTy
+//
+// Returns:
+//   - string: The download URL for the exported flow
+//   - error: An error if the URL generation process fails
+//
+// The function performs the following steps:
+// 1. Creates an export job for the specified flow
+// 2. Polls the export job status until completion or timeout
+// 3. Returns the download URL when the job completes successfully
+//
+// It will return an error if:
+//   - The export job creation fails
+//   - The job polling times out
+//   - The job fails to generate a download URL
+func (a *architectFlowProxy) generateDownloadUrl(flowId string) (string, error) {
+	return a.generateDownloadUrlAttr(a, flowId)
+}
+
+// createExportJob creates an export job for a specified architect flow.
+//
+// Parameters:
+//   - a: *architectFlowProxy - The architect flow proxy instance
+//   - flowId: string - The ID of the flow to be exported
+//
+// Returns:
+//   - string: The ID of the created export job
+//   - *platformclientv2.APIResponse: The raw API response
+//   - error: An error if the job creation fails or if the response is invalid
+//
+// The function creates an export job by making a POST request to the flows export jobs endpoint.
+// It will return an error if:
+//   - The API call fails
+//   - The created job response is nil
+//   - The job ID in the response is nil
+func (a *architectFlowProxy) createExportJob(flowId string) (string, *platformclientv2.APIResponse, error) {
+	return a.createExportJobAttr(a, flowId)
+}
+
+// getExportJobStatusById retrieves the status of an architect flow export job.
+// Implementation function: getExportJobStatusByIdFn
+// Parameters:
+//   - jobId: string - The ID of the export job to check
+//
+// Returns:
+//   - *platformclientv2.ArchitectFlowJobStatusResponse: The job status response
+//   - *platformclientv2.APIResponse: The raw API response
+//   - error: An error if the job status cannot be retrieved or if the response is invalid
+//
+// The function will return an error if the job status or status field is nil in the response.
+func (a *architectFlowProxy) getExportJobStatusById(jobId string) (*platformclientv2.Architectexportjobstateresponse, *platformclientv2.APIResponse, error) {
+	return a.getExportJobStatusByIdAttr(a, jobId)
+}
+
+// pollExportJobForDownloadUrl polls an architect flow export job until it completes or times out.
+// It checks the job status periodically and waits for the download URL to become available.
+// Implementation function: pollExportJobForDownloadUrlFn
+//
+// Parameters:
+//   - a: *architectFlowProxy - The architect flow proxy instance
+//   - jobId: string - The ID of the export job to poll
+//
+// Returns:
+//   - string: The download URL for the exported flow
+//   - error: An error if the polling times out or if there's an issue getting the job status
+//
+// The function will timeout after 90 seconds if the job doesn't complete.
+// It polls the job status every 1 second.
```

Comment:
Thanks for putting all these docs in for these functions. It will help with previewing the functions in the IDE. Good work @charliecon !